lib/shanty_properties_file_plugin/encrypted_properties.rb

```diff
@@ -0,0 +1,33 @@
+require 'gpgme'
+require 'shanty_properties_file_plugin/properties'
+
+module PropertiesFilePlugin
+  # Public: properties class for getting environment config from properties files
+  class EncryptedProperties < Properties
+    # Public: initialize the properties class
+    #
+    # project_path - path to the project directory
+    # default_file - name of the default properties files
+    # files        - properties files to be processed
+    def initialize(project_path, default_file, files, passphrase)
+      @project_path = project_path
+      @default_file = default_file
+      @files = files
+      @passphrase = passphrase
+      @crypto = GPGME::Crypto.new
+    end
+
+    protected
```

Comment:
Not sure you want `protected` for this method. In Ruby, `protected` means something really weird which is all about the receiver object. Basically, if you want the `protected` style from Java/Scala/C# `private` does the trick as you can inherit `private` methods in Ruby:

```ruby
class Foo
  private
  
  def foo
    'foo'
  end
end

class Bar < Foo
  def bar
    foo
  end
end

# Note how this doesn't raise an exception like it would in Java.
Bar.new.bar # => 'foo'
```

Methods marked `protected` in Ruby are exactly the same as `private` except that they can be called externally when `self` is of the same type as the definition location:

```ruby
class Foo
  protected
  
  def foo
    'foo'
  end
end

Foo.new.foo # => NoMethodError: protected method `foo' called

class Bar < Foo
  def compare(o)
    # Note how we're technically calling this like a public method but it's protected. This works because the call site
    # is within a class or subclass of the method definition we're calling.
    self.foo == o.foo
  end
end

Bar.new.compare(Foo.new) # => true
```